narayana/ArjunaJTA/jta/tests/classes/com/arjuna/ats/jta/xa/performance/JTAStoreTests.java

```diff
@@ -40,40 +40,29 @@
 @Threads(JMHConfigJTA.BT)
 @State(Scope.Benchmark)
 public class JTAStoreTests {
-    @State(Scope.Benchmark)
-	public static class BenchmarkState {
-        TransactionManager tm;
-
-		public BenchmarkState() {
-          tm = com.arjuna.ats.jta.TransactionManager.transactionManager();
-          try {
-              BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class).setNodeIdentifier("0");
-          } catch (CoreEnvironmentBeanException e) {
-              throw new RuntimeException(e);
-          }
-
-          BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).
-              setObjectStoreType(VolatileStore.class.getName());
-		}
-    }
-
-    @Setup(Level.Trial)
-    public void before() {
+    static {
+      try {
+        BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class).setNodeIdentifier("0");
+        BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).setObjectStoreType(VolatileStore.class.getName());
+        tm = com.arjuna.ats.jta.TransactionManager.transactionManager();
+      } catch (Exception e) {
+          throw new RuntimeException(e);
+      }
     }
 
-    @TearDown(Level.Trial)
-    public void after() {
-    }
+    private static TransactionManager tm;
+    private XAResourceImpl resource1 = new XAResourceImpl();
+    private XAResourceImpl resource2 = new XAResourceImpl();
 
     @Benchmark
-    public boolean jtaTest(BenchmarkState state) {
+    public boolean jtaTest() {
         try {
-            state.tm.begin();
+            tm.begin();
 
-            state.tm.getTransaction().enlistResource(new XAResourceImpl());
-            state.tm.getTransaction().enlistResource(new XAResourceImpl());
+            tm.getTransaction().enlistResource(resource1);
+            tm.getTransaction().enlistResource(resource2);
 
-            state.tm.commit();
+            tm.commit();
         } catch(Exception e) {
             System.err.printf("JTAStoreTests#jtaTest: %s%n", e.getMessage());
```

Comment:
FYI the reason I don't return false here is because the returned value is not important. If the method was void there would be a risk that the compiler would optimise away the whole method since the result never gets used. Maybe JMH does something clever to ensure that this does not happen but returning an arbitrary value does no harm.